app/components/createContainer/createContainerController.js

```diff
@@ -231,6 +241,13 @@ function ($scope, $state, $stateParams, $filter, Config, Info, Container, Contai
       networkMode += ':' + containerName;
     }
     config.HostConfig.NetworkMode = networkMode;
+    var extraHosts = [];
```

Comment:
Use the `formValues.ExtraHosts` as your temporary object instead of this local variable. So you can remove this variable, iterate on each entry in `formValues.ExtraHosts` and directly push correct values to `config.HostConfig.ExtraHosts`.